subcommands/config/log.go

```diff
@@ -30,18 +30,52 @@ func doConfigLog(cmd *cobra.Command, args []string) {
 		subcommands.LogConfigs(&subcommands.LogConfigsOptions{
 			Limit: listLimit,
 			ListFunc: func() (*client.DeviceConfigList, error) {
-				return api.FactoryListConfig(factory)
+
+				list, err := api.FactoryListConfig(factory)
+				for _, dc := range list.Configs {
+					if len(dc.CreatedBy) > 0 {
+						user, _ := api.UserAccessDetails(factory, dc.CreatedBy)
+						dc.CreatedBy = user.Name
```

Comment:
I wonder if we _really_ want to show a user name?

In some places like this we show just user IDs, and that is good enough.
That way all my other _optimization_ comments become irrelevant.

But, if there is a good reason to show a user name - I'd also show a user ID and/or email.
Both attributes provide a way more valuable debugging information to a seasoned admin.

Reply:
I think we can do this in a better way that addresses performance and also doesn't make users have to constantly look up uid's. We have an api `UsersList` which looks up all the users in a factory in one call returning an array like:
```
{
        "polis-id": "59db9c9a1c85010019e023cc",
        "name": "Andy Doan",
        "role": "owner",
        "email": "andy@foundries.io"
    },
```

There's a few ways to do it. One is we pass this array to the LogConfigOptions. Then that function can do a lookup *and* I think it removes a bunch of copy/paste we have in this PR

Reply:
@doanac isn't a list API paging?

I mean, as we grow - there would be no wonder to have a factory with 5000 users.
At that size, the list-all API becomes non-scalable; a get one-by-one is easier scale as we have an eternal hard limit here.

P.S. Even better would be to have an API which accepts a list of user IDs and returns objects.
But, at our current stage I see no need for that; this task can be done without API changes.

Reply:
I don't believe the API paginates. Its one of the ones we've concluded can function fine even with large numbers.

Reply:
If no pagination - we can give it a try.

Reply:
Though I'm not sure if it will save a lot of code.

Duplication here is easily tackled; so it's a matter of _how many lines of code to have in one place_.